xbmc/pictures/Picture.cpp

```diff
@@ -253,20 +253,21 @@ bool CPicture::CacheTexture(uint8_t *pixels, uint32_t width, uint32_t height, ui
     uint32_t dest_width_aligned = ((dest_width + 15) & ~0x0f);
     uint32_t stride = dest_width_aligned * sizeof(uint32_t);
 
-    uint32_t* buffer = new uint32_t[dest_width_aligned * dest_height + 4];
+    auto buffer = std::make_unique<uint32_t[]>(dest_width_aligned * dest_height + 4);
     if (buffer)
```

Comment:
This `if` statement isn't required, buffer will never be empty.

Reply:
I added an additional commit that removes all the unnecessary nullptr checks from the file.

Reply:
thanks!